app/controllers/self_service/trading_names_controller.rb

```diff
@@ -15,6 +15,20 @@ def create
       end
     end
 
+    def edit
+      @firm = principal.firm.subsidiaries.find(params[:id])
```

Comment:
we should stick to `principal.firm.trading_names` here — maybe change the usage in `initialize_firm_from_lookup_trading_name` too while we're here